blocks/product-comparison-table/product-comparison-table.js

```diff
@@ -0,0 +1,101 @@
+import getMockData from './product-mock-data.js';
+
+const fakeData = getMockData();
+
+function buildPriceContainer(productName, numberOfDevices, elementToReplace) {
+  const priceContainer = document.createElement('div');
+  priceContainer.classList.add('product-comparison-price');
+
+  const productData = fakeData
+    .filter((product) => product.data.product.product_name === productName);
+  if (productData.length === 0) {
+    return;
+  }
+  const variationPrice = productData[0].data.product.variations[numberOfDevices][1];
+  const productVariationPrice = variationPrice.price;
+  const productVariationDiscountPrice = variationPrice.discount.discounted_price;
+  const priceLabel = variationPrice.currency_label;
+
+  if (productVariationDiscountPrice) {
+    const oldPriceContainer = document.createElement('div');
+    oldPriceContainer.classList.add('old-price-container');
+    oldPriceContainer.innerHTML = `<p>Old Price <span class='price-strike-through'>${productVariationPrice} ${priceLabel}</span></p>`;
+    priceContainer.appendChild(oldPriceContainer);
+  }
+
+  const currentPriceContainer = document.createElement('div');
+  currentPriceContainer.classList.add('current-price-container');
+  currentPriceContainer.innerHTML = `<p>${productVariationDiscountPrice ?? productVariationPrice} ${priceLabel}</p>`;
+  priceContainer.appendChild(currentPriceContainer);
+
+  elementToReplace.replaceWith(priceContainer);
+}
+
+export default function decorate(block) {
+  block.setAttribute('role', 'table');
+
+  block.querySelectorAll('div')
+    .forEach(async (div) => {
+      if (div.children.length > 1) {
+        div.setAttribute('role', 'row');
+      } else if (div.children.length <= 1 && !div.hasAttribute('role')) {
+        div.setAttribute('role', 'cell');
+      }
+      if (div.firstElementChild?.tagName === 'STRONG'
+        || div.firstElementChild?.firstElementChild?.tagName === 'STRONG') {
+        div.classList.add('active');
+      }
+      if (div.textContent.match(/^[yY]es/)) {
```

Comment:
`/i` could be used for case insensitive search

Reply:
updated